pkg/snapshot/util/gzip.go

```diff
@@ -0,0 +1,56 @@
+package util
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io"
+)
+
+var GzipMagicHeader = []byte{'\x1f', '\x8b'}
+
+func NewGzipReadCloser(r io.ReadCloser, level int) io.ReadCloser {
+	pr, pw := io.Pipe()
+	go func() {
+		defer r.Close()
+
+		gw, err := gzip.NewWriterLevel(pw, level)
+		if err != nil {
+			pw.CloseWithError(err)
+			return
+		}
+
+		if _, err := io.Copy(gw, r); err != nil {
+			gw.Close()
+			pw.CloseWithError(err)
+			return
+		}
+		gw.Close()
```

Comment:
Does the return error on the gzip writer's Close need to be checked? I don't know if it matters since the Write error is checked, but just wondering.

Reply:
I am thinking it should be checked since I'm looking at it now and the `Close()` appears that it can definitely produce an error when it does the final call to `Write()` to get write the gzip footer or possibly when it closes the `flate.Writer`.

Reply:
Ok, I am now checking the `gw.Close` error and I also removed the `gw.Close()`  that was being called when there was an error caused by `io.Copy` since it should not attempt to finish writing the gzip file for any of the possible error outcomes (which are like unexpected EOF, short write, etc).